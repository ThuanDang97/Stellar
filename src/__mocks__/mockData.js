export const TOOLTIP_LOREM =
  'Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.'

export const SELECT_OPTIONS = [
  { value: '1', name: 'Pay monthly' },
  { value: '3', name: 'Pay 3 monthly' },
  { value: '6', name: 'Pay 6 monthly' },
  { value: '12', name: 'Pay annually' },
]

export const USER_FULL_NAME = 'John Wick'

export const AVATAR_DEFAULT = {
  firstName: 'John',
  lastName: 'Wick',
}

export const AVATAR_IMAGE = {
  imageUrl: 'https://picsum.photos/20',
}

export const NETFLIX_REPORT = {
  name: 'netflix',
  period: 'report monthly',
}

export const GOAL_ITEM = {
  title: 'Improve My Credit Score',
  iconUrl: '/icons/piggy-bank',
}

export const BANK_NUMBER = '123456789111'

export const BANK_NAME = 'Chase'

export const INPUT_VALUES = {
  email: 'test@gmail.com',
  text: 'example',
  password: 'example1password',
  image:
    'data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2716%27%20height=%2716%27/%3e',
}

export const MEDIA_IMAGES = {
  netflix: '/images/netflix.png',
  hulu: '/images/hulu.png',
}

export const ADD_BILL_LIST = [
  {
    id: '1',
    description: 'Netflix',
    amount: 50,
    buttonControl: '',
    imageUrl: MEDIA_IMAGES.netflix,
    created: 121212121,
  },
  {
    id: '2',
    description: 'Holo',
    amount: 77,
    buttonControl: '',
    imageUrl: MEDIA_IMAGES.hulu,
    created: 121212121,
  },
]

export const MY_GOALS = [
  {
    id: '1',
    goal: 'Buy a car',
    scoreNeeded: 703,
    pointNeeded: 'READY',
    imageUrl: MEDIA_IMAGES.netflix,
    created: 121212121,
  },
  {
    id: '2',
    goal: 'Buy a house',
    scoreNeeded: 720,
    pointNeeded: '23 PTS AWAY',
    imageUrl: MEDIA_IMAGES.hulu,
    created: 3434543645,
  },
]

export const LINKED_BILL_LIST = [
  {
    id: '1',
    description: 'Netflix',
    amount: 50,
    draftDate: '1st',
    status: 'Unpaid',
    imageUrl: MEDIA_IMAGES.netflix,
    created: 121212121,
  },
  {
    id: '2',
    description: 'Hulu',
    amount: 32,
    draftDate: '4th',
    status: 'Paid',
    imageUrl: MEDIA_IMAGES.hulu,
    created: 121212121,
  },
]

export const EXITS_EMAIL = 'thanh@endurance.com'

export const NUMBER_BANK = [
  {
    id: '1',
    bankNumber: '12435324234',
    bankName: 'Chase',
  },
]

export const ACCOUNT_LOGIN = [
  {
    userId: '1',
    email: 'khanhtam304n@gmail.com',
    password: 'Khanhtam@123',
  },
  {
    userId: '2',
    email: 'khanhtam3041n@gmail.com',
    password: 'Khanhtam@123',
  },
]

export const USER = {
  userId: '4',
  userName: 'Khanh Tam',
  email: 'khanhtam304n@gmail.com',
  password: 'Khanhtam@123',
}

export const GOAL = ['g1', 'g2']

export const MARKETING = ['m1', 'm2']

export const USER_ID = '123'

export const DATE_TIMER = new Date('08/28/2022')

export const USER_GOAL = { userId: USER_ID, goals: GOAL }

export const USER_MARKETING = { userId: USER_ID, goals: MARKETING }

export const ADDRESS = {
  address: '123 street',
  phone: '1231231231',
}

export const SSN_NUMBER = '123123123'

export const SUBSCRIPTION_PLAN = 'flex'

export const USER_SUBSCRIPTION_PLAN = {
  userId: USER_ID,
  subscriptionPlan: SUBSCRIPTION_PLAN,
}

export const CHART_SCORE = {
  data: [
    { month: 'M', score: 720 },
    { month: 'A', score: 740 },
    { month: 'M', score: 768 },
    { month: 'J', score: 785 },
    { month: 'J', score: 810 },
    { month: 'A', score: 830 },
  ],
}

export const LINK_TOKEN = 'sand-box-123'

export const BANK_ID = 'id_45'

export const BANK_INFO = {
  id: BANK_ID,
  bankNumber: BANK_NUMBER,
  bankName: BANK_NAME,
}
export const FICO_DATE_UPDATE = 'Last Updated Aug, 12, 2021'

export const NOTIFICATIONS = [
  {
    id: '1',
    title: 'Credit Score Update',
    desc: 'Your credit score increased by...',
    dateTime: '1 hours',
    isCheck: false,
  },
  {
    id: '2',
    title: 'Bill Pay',
    desc: 'We noticed you missed a payment...',
    dateTime: '3 days ago',
    isCheck: false,
  },
  {
    id: '3',
    title: 'Offer',
    desc: 'Look like you may quality for...',
    dateTime: '10 days ago',
    isCheck: false,
  },
  {
    id: '4',
    title: 'Offer',
    desc: 'Look like you may quality for offer',
    dateTime: '11 days ago',
    isCheck: false,
  },
]

export const APPLICATION_INFO = {
  title: 'Netflix',
  image: '/images/netflix.png',
}

export const APPLICATION_CONNECT_ACCOUNT = {
  userId: '1',
  email: 'khanhtam304n@gmail.com',
  password: 'Khanhtam@123',
}

// Min score: 300 - Max score 850
export const FICO_SCORE = 500

export const IDENTITY = {
  firstName: 'John',
  lastName: 'Wick',
  dateOfBirth: '02/02/2000',
  ssn: 123123123,
}

export const SSN_USER = {
  userId: USER_ID,
  ssn: SSN_NUMBER,
}

export const USER_IDENTITY = {
  userId: USER_ID,
  identity: IDENTITY,
}

export const ADDRESS_USER = {
  address: '604 Nui Thanh',
  phoneNumber: '+10949328492',
}

export const FICO_UNLOCK_SCORE = 708

export const LIST_BANKS_NUMBER = [
  { id: 'Chase', bankNumber: '423432432423232' },
  { id: 'Bank of America', bankNumber: '111122223333000' },
  { id: 'CitiBank Online', bankNumber: '111122223333111' },
]

export const INFO_PAYMENTS = {
  id: '123',
  money: '15.99',
  bankName: 'Chase',
  payDate: new Date(),
  numberPaymentsMonth: '3',
  appInfo: {
    title: 'Netflix',
    image: '/images/netflix.png',
  },
  status: 'UNPAID',
}

export const BILL_INFO = {
  title: 'Netflix',
  amount: 10,
}

export const ADDED_BILL_ID = '1234'

export const USER_DETAIL = {
  id: '123',
  amount: 10,
  draftDate: 31,
  bankNumber: BANK_NUMBER,
}

export const EDIT_LINK_BILL = {
  id: '123',
  amount: 10,
}
