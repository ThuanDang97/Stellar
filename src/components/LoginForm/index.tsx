import React from 'react'
import { useTheme } from 'styled-components'

// Props type
import { LoginFormProps } from '@self-types/components/LoginForm.props'

// Components
import Button from '@components/Button'
import Input from '@components/Input'
import ErrorMessage from '@components/ErrorMessage'
import { BoxShadowStyled } from '@components/styled-components'
import LinkComponent from '@components/Link'

// Constants
import { DEFAULT_HEADER_URL } from '@constants/routes'

const LoginForm = ({
  errorMessage,
  handleChangeEmail,
  handleChangePassword,
  email,
  password,
  onSubmit,
}: LoginFormProps) => {
  const theme = useTheme()

  return (
    <BoxShadowStyled
      data-testid="loginForm"
      display="flex"
      flexDirection="column"
      alignItems="stretch"
      as="form"
      noValidate
      onSubmit={onSubmit}
      padding={theme.metrics.dimensions.md}
    >
      <Input
        mTop={theme.metrics.dimensions.md}
        mBottom={theme.metrics.dimensions.md}
        name="email"
        type="email"
        placeholder="Enter your email address"
        value={email}
        iconUrl="/icons/envelope"
        onChange={handleChangeEmail}
        error={errorMessage?.email}
      />
      <Input
        mTop={theme.metrics.dimensions.sm}
        name="password"
        type="password"
        isLogin
        iconUrl="/icons/icon-lock"
        placeholder="Password"
        value={password}
        error={errorMessage?.password}
        onChange={handleChangePassword}
      />
      {(errorMessage?.email || errorMessage?.password) && (
        <ErrorMessage error={errorMessage?.email || errorMessage?.password} />
      )}
      <Button
        title="Login"
        type="submit"
        disabled={!(email && password)}
        mTop={theme.metrics.dimensions.xlg}
      />
      <LinkComponent
        href={DEFAULT_HEADER_URL.FORGOT_PASSWORD.URL}
        linkTypes="forgotPasswordLink"
        mRight={theme.metrics.dimensions.sm}
        text="Forgot Password?"
        pTop={theme.metrics.dimensions.sm}
        pBottom={theme.metrics.dimensions.tiny}
      />
    </BoxShadowStyled>
  )
}

export default React.memo(LoginForm)
