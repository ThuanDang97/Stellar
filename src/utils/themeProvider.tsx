import { ThemeProvider } from 'styled-components'
import { render } from '@testing-library/react'
import { theme } from '@themes/index'

export const renderWithTheme = (component: React.ReactNode) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
}
