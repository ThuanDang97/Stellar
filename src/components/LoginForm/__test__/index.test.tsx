import React from 'react'
import { screen, cleanup, fireEvent } from '@testing-library/react'

// Mock data
import { INPUT_VALUES } from '@mocks/mockData'

// Utils
import { renderWithTheme } from '@utils/themeProvider'

// Components
import LoginForm from '../index'

describe('LoginForm', () => {
  const props = {
    handleChangeEmail: jest.fn(),
    handleChangePassword: jest.fn(),
    onSubmit: jest.fn(),
    email: '',
    password: '',
  }

  beforeEach(() => {
    renderWithTheme(<LoginForm {...props} />)
  })

  afterEach(() => {
    cleanup()
  })

  it('should render input email', () => {
    const emailInput = screen.getAllByTestId('iconInput')

    expect(emailInput[0]).toBeInTheDocument()
  })

  it('should call onChange input email', () => {
    const input = screen.getAllByTestId('iconInput')

    fireEvent.change(input[0], {
      target: { value: INPUT_VALUES.email },
    })

    expect(props.handleChangeEmail).toBeCalled()
  })

  it('should render input password', () => {
    const passwordInput = screen.getAllByTestId('iconInput')
    expect(passwordInput[1]).toBeInTheDocument()
  })

  it('should call onChange input password', () => {
    const input = screen.getAllByTestId('iconInput')

    fireEvent.change(input[1], {
      target: { value: INPUT_VALUES.email },
    })

    expect(props.handleChangePassword).toBeCalled()
  })

  it('should call submit form', () => {
    const button = screen.getByTestId('button')

    fireEvent.submit(button)

    expect(props.onSubmit).toBeCalled()
  })

  it('should render button submit', () => {
    const button = screen.getByTestId('button')

    expect(button).toBeInTheDocument()
  })

  it('should render button disable submit', () => {
    const button = screen.getByTestId('button')

    expect(button).toHaveAttribute('disabled')
  })

  it('should render button enable', () => {
    cleanup()

    const newProps = {
      ...props,
      email: 'test',
      password: 'test',
    }

    renderWithTheme(<LoginForm {...newProps} />)

    const button = screen.getByTestId('button')

    expect(button).not.toHaveAttribute('disabled')
  })

  it('should render ErrorMessage', () => {
    const newProps = {
      ...props,
      errorMessage: { email: 'Invalid', password: 'Invalid' },
      email: '',
      password: '',
    }

    renderWithTheme(<LoginForm {...newProps} />)

    const ErrorMessage = screen.getByTestId('errorMessage')

    expect(ErrorMessage).toBeInTheDocument()
  })

  it('matches snapshot', () => {
    const { asFragment } = renderWithTheme(<LoginForm {...props} />)

    expect(asFragment()).toMatchSnapshot()
  })
})
