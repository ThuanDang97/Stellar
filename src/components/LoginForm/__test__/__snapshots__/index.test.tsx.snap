// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 kjqdjo eFQizc"
    data-testid="loginForm"
    display="flex"
    novalidate=""
  >
    <div
      class="ViewStyled-sc-rqe5sa-0 InputStyled__WrapperInputStyled-sc-9gj86m-1 UsSdI brcjID"
      color="#203F59"
      data-testid="inputIcon"
      display="block"
    >
      <div
        class="ViewStyled-sc-rqe5sa-0 InputStyled__WrapperIconStyled-sc-9gj86m-2 fTsvHV kHphsF"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
        >
          <img
            alt="icon"
            data-nimg="fill"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
      <input
        class="InputStyled-sc-1k09cqb-0 InputStyled__InputItemStyled-sc-9gj86m-0 hnxmhM NDCRX"
        data-testid="iconInput"
        name="email"
        placeholder="Enter your email address"
        type="email"
        value=""
      />
    </div>
    <div
      class="ViewStyled-sc-rqe5sa-0 InputStyled__WrapperInputStyled-sc-9gj86m-1 bNeJX brcjID"
      color="#203F59"
      data-testid="inputIcon"
      display="block"
    >
      <div
        class="ViewStyled-sc-rqe5sa-0 InputStyled__WrapperIconStyled-sc-9gj86m-2 fTsvHV kHphsF"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
        >
          <img
            alt="icon"
            data-nimg="fill"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
      <input
        class="InputStyled-sc-1k09cqb-0 InputStyled__InputItemStyled-sc-9gj86m-0 hnxmhM NDCRX"
        data-testid="iconInput"
        name="password"
        placeholder="Password"
        type="password"
        value=""
      />
    </div>
    <button
      class="ButtonStyled-sc-1ys9v52-0 iAfDLq"
      data-testid="button"
      disabled=""
      type="submit"
    >
      Login
    </button>
    <a
      class="LinkStyled__ButtonLinkStyled-sc-626068-1 hvsWbe"
      data-testid="link"
      href="/forgot-password"
    >
      Forgot Password?
    </a>
  </form>
</DocumentFragment>
`;
